src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,76 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  filter: Filter;
+  setFilter: (prop: Filter) => void;
+  query: string;
+  setQuery: (prop: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filter,
+  setFilter,
+  query,
+  setQuery,
+}) => {
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case Filter.All:
+        setFilter(Filter.All);
+        break;
+
+      case Filter.Active:
+        setFilter(Filter.Active);
+        break;
+
+      case Filter.Completed:
+        setFilter(Filter.Completed);
+        break;
+
+      default:
+        break;
+    }
```

Comment:
actually looks like you can do just
```suggestion
    setFilter(event.target.value)
```